Leeroy Jenkins PR

neerpatel opened the pull request:
This pull request includes several changes to the `app/components`, `app/pages`, and `app/package.json` files to improve the layout and styling of the application. The most important changes involve replacing CSS modules with Tailwind CSS classes, updating the `Footer` and `NavBar` components, and adding new dependencies.

### Layout and Styling Improvements:

* [`app/components/Footer.tsx`](diffhunk://#diff-65edf535b119e7cb1b653d5546abcf5d2f3d35d580ce5d54c63574ba22533862L2-L15): Replaced `react-bootstrap/Navbar` with a custom footer layout using Tailwind CSS classes. Added links to the privacy policy and GitHub.
* [`app/components/NavBar.tsx`](diffhunk://#diff-1492eb22f76cbf14526dca503504559b27670c67976f73747ec88ca6cfcfc084L4-R6): Updated the `NavBar` component to use Tailwind CSS for styling, included a new logo, and removed redundant CSS module imports. [[1]](diffhunk://#diff-1492eb22f76cbf14526dca503504559b27670c67976f73747ec88ca6cfcfc084L4-R6) [[2]](diffhunk://#diff-1492eb22f76cbf14526dca503504559b27670c67976f73747ec88ca6cfcfc084R42-R98)
* [`app/pages/404.tsx`](diffhunk://#diff-f9af9e3b32d0a94c881914c515a884d7c32155879976f21e9c2ece56af2948c5L2-R3): Refactored the 404 error page to use Tailwind CSS for styling and removed `react-bootstrap` components and CSS module imports. [[1]](diffhunk://#diff-f9af9e3b32d0a94c881914c515a884d7c32155879976f21e9c2ece56af2948c5L2-R3) [[2]](diffhunk://#diff-f9af9e3b32d0a94c881914c515a884d7c32155879976f21e9c2ece56af2948c5L16-R37)

### Dependency Updates:

* [`app/package.json`](diffhunk://#diff-012b982f97a0d326aeec58dd3b789484b05a944d609afe1b8ed5e299fc485f61R24-R39): Added new dependencies for `react-icons`, `autoprefixer`, `sass`, and `tailwindcss-3d`.

### CSS Module Removal:

* [`app/components/ProfileCard.tsx`](diffhunk://#diff-1c65d3168c3bf2adeeb61d881839e4d95dc950666bdcec3916b40788fd81555eL1): Removed CSS module imports and replaced class names with Tailwind CSS classes. [[1]](diffhunk://#diff-1c65d3168c3bf2adeeb61d881839e4d95dc950666bdcec3916b40788fd81555eL1) [[2]](diffhunk://#diff-1c65d3168c3bf2adeeb61d881839e4d95dc950666bdcec3916b40788fd81555eL15-R24)
* [`app/pages/ai-gen.tsx`](diffhunk://#diff-ee8da3fb2f1ad18053af53b32ead5760161ec04a08e19375b9d85f583c8a7364L17): Removed CSS module imports and replaced class names with Tailwind CSS classes. [[1]](diffhunk://#diff-ee8da3fb2f1ad18053af53b32ead5760161ec04a08e19375b9d85f583c8a7364L17) [[2]](diffhunk://#diff-ee8da3fb2f1ad18053af53b32ead5760161ec04a08e19375b9d85f583c8a7364L184-R191)
* [`app/pages/gallery.tsx`](diffhunk://#diff-c91b6a2ff1050b9b081abe3045ea192ce5bdd772234f6cb4a30d1ae64e06a029L4): Removed CSS module imports and replaced class names with Tailwind CSS classes. [[1]](diffhunk://#diff-c91b6a2ff1050b9b081abe3045ea192ce5bdd772234f6cb4a30d1ae64e06a029L4) [[2]](diffhunk://#diff-c91b6a2ff1050b9b081abe3045ea192ce5bdd772234f6cb4a30d1ae64e06a029L121-R121) [[3]](diffhunk://#diff-c91b6a2ff1050b9b081abe3045ea192ce5bdd772234f6cb4a30d1ae64e06a029L185-R198) [[4]](diffhunk://#diff-c91b6a2ff1050b9b081abe3045ea192ce5bdd772234f6cb4a30d1ae64e06a029L215-R214)
* `app/pages/gallery/image/[id].tsx`: Removed CSS module imports and replaced class names with Tailwind CSS classes. ([app/pages/gallery/image/[id].tsxL6](diffhunk://#diff-0c85ea45a9abb5942cf2aa2fa2944f592caf3799e0f6a8f25598bc1c2bf5e8c2L6), [app/pages/gallery/image/[id].tsxL55-R54](diffhunk://#diff-0c85ea45a9abb5942cf2aa2fa2944f592caf3799e0f6a8f25598bc1c2bf5e8c2L55-R54), [app/pages/gallery/image/[id].tsxL111-R118](diffhunk://#diff-0c85ea45a9abb5942cf2aa2fa2944f592caf3799e0f6a8f25598bc1c2bf5e8c2L111-R118))

neerpatel committed:
Add react-icons package and update package-lock.json

neerpatel committed:
Refactor Footer component to use Link component for Privacy Policy

neerpatel committed:
Refactor CSS files and package dependencies

neerpatel committed:
Merge branch 'main' into ndp/footer_update

rm-potato commented on app/components/NavBar.tsx:
```diff
@@ -43,53 +39,64 @@ const NavBar: React.FC = () => {
-                  color: radio.value === 'dark' ? (isDarkTheme ? 'white' : '#2d345b') : (isDarkTheme ? '#e1a629' : 'white'),
-                  borderColor: radio.value === 'dark' ? '#2d345b' : '#e1a629'
-                }}
+                className="mx-1"
```
curious how you plan on handling this double check for color styling? tried moving it to a css file myself and had a heck of time with it.

neerpatel committed:
Refactor NavBar theme toggle styles

neerpatel committed:
Refactor Footer component and 404 page

- Refactor Footer component to update link behavior and add GitHub icon
- Update 404 page to use react-bootstrap/Image and display custom image

neerpatel committed:
Refactor NavBar theme toggle styles
